From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martijn Muijsers <martijnmuijsers@live.nl>
Date: Mon, 10 Jul 2023 22:05:08 +0200
Subject: [PATCH] Modifiable Bukkit enums - Remove hard-coded properties -
 Material - Key

Adds a test for equality between `Material.getKey` and its original backing implementation.

License: GPL-3.0 (https://www.gnu.org/licenses/gpl-3.0.html)
Fiddle - https://fiddlemc.org

diff --git a/src/test/java/org/fiddlemc/fiddle/material/legacy/PerNonLegacyMaterialTest.java b/src/test/java/org/fiddlemc/fiddle/material/legacy/PerNonLegacyMaterialTest.java
index f11b7d202160a598ab2d431960d02791f3abcb28..a7e3252b9e4c1d8a218500adc58744335cc636e3 100644
--- a/src/test/java/org/fiddlemc/fiddle/material/legacy/PerNonLegacyMaterialTest.java
+++ b/src/test/java/org/fiddlemc/fiddle/material/legacy/PerNonLegacyMaterialTest.java
@@ -35,6 +35,15 @@ public class PerNonLegacyMaterialTest extends AbstractTestingBase {
         }
     }
 
+    // Fiddle start - modifiable Bukkit enums - Material - remove hard-coded properties - key
+    @Test
+    public void getKey() {
+        for (Material material : getMaterials()) {
+            assertEquals(org.bukkit.NamespacedKey.minecraft(material.name().toLowerCase(java.util.Locale.ROOT)), material.getKey(), "getKey is incorrect for " + material);
+        }
+    }
+    // Fiddle end - modifiable Bukkit enums - Material - remove hard-coded properties - key
+
     // Fiddle start - modifiable Bukkit enums - Material - remove hard-coded properties - max stack size
     @Test
     public void getMaxStackSize() {
